airflow/jobs/local_task_job.py

```diff
@@ -208,6 +208,13 @@ def heartbeat_callback(self, session=None):
                 )
                 raise AirflowException("PID of job runner does not match")
         elif self.task_runner.return_code() is None and hasattr(self.task_runner, 'process'):
+            if ti.state == State.SKIPPED:
+                # A DagRun timeout will cause tasks to be externally marked as skipped.
+                dagrun = ti.get_dagrun()
+                execution_time = (dagrun.end_date or timezone.utcnow()) - dagrun.start_date
+                dagrun_timeout = ti.task.dag.dagrun_timeout
```

Comment:
Is there any chance that the value of `ti.task` or `ti.task.dag` is not set? I'm wondering if this requires additional error handling.

Reply:
I think we _should_ be good as LocalTaskJob is only created when running a task with the following flow - 

https://github.com/apache/airflow/blob/4faf9adf4e29823d2feeecdd1cc8699b27460a5d/airflow/jobs/local_task_job.py#L40-L48

https://github.com/apache/airflow/blob/4faf9adf4e29823d2feeecdd1cc8699b27460a5d/airflow/cli/commands/task_command.py#L151-L152

https://github.com/apache/airflow/blob/4faf9adf4e29823d2feeecdd1cc8699b27460a5d/airflow/cli/commands/task_command.py#L105

https://github.com/apache/airflow/blob/4faf9adf4e29823d2feeecdd1cc8699b27460a5d/airflow/cli/commands/task_command.py#L299

https://github.com/apache/airflow/blob/4faf9adf4e29823d2feeecdd1cc8699b27460a5d/airflow/cli/commands/task_command.py#L282-L289

which has `dag` as well as `task`

Reply:
Fantastic, thanks for the context.